packages/hydrogen/src/customer/auth.helpers.ts

```diff
@@ -0,0 +1,242 @@
+import {BadRequest} from './BadRequest';
+
+export const userAgent =
+  'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.93 Safari/537.36';
+
+export interface HydrogenSession {
+  get: (key: string) => string | undefined;
+  set: (key: string, value: string) => void;
+  unset: (key: string) => void;
+  commit: () => Promise<string>;
+}
```

Comment:
Yeah, I wasn't sure what to do about that. I imagine most people take our default cookie session implementation, but I'm willing to bet other people use an actual DB of some sort to store sessions 🤔

Reply:
They could still extend our class and overwrite the `constructor` / etc. behavior? 🤔 
Ah, maybe we wouldn't tree-shake the cookie related stuff in that case? So perhaps better to have an abstract `HydrogenSession` class and another `HydrogenCookieSession`? Well, just random ideas.

Reply:
I'd like for people to be able to use any of the [existing remix session ](https://remix.run/docs/en/main/utils/sessions)utils easily. But it's looking like Remix 3 will be more opinionated about sessions. Maybe we should also be more opinionated about sessions? It would actually be pretty cool if we offered a default non-cookie based solution. For down the road though.